lcls_tools/common/devices/reader.py

```
@@ -142,6 +143,24 @@ def create_lblm(
         return LBLMCollection(**device_data)
 
 
+def create_bpm(
+    area: str = None, name: str = None
+) -> Union[None, BPM, BPMCollection]:
+    device_data = _device_data(area=area, device_type="bpms", name=name)
+    if not device_data:
+        return None
+    if name:
+        try:
+            # this data is not available from YAML directly in this form, so we add it here.
+            device_data.update({"name": name})
+            return BPM(**device_data)
+        except ValidationError as field_error:
```

In a similar vain to the above comment, I think we should just pass the exception here, maybe with a bespoke error message. If the caller needs to ignore the error, they can do so with their own try/catch block.

---

Also in support of changing exception-passing to something more intuitive.